gtsam_unstable/timing/timeShonanAveraging.cpp

```diff
@@ -79,9 +79,9 @@ void saveG2oResult(string name, const Values& values, std::map<Key, Pose3> poses
         myfile << "VERTEX_SE3:QUAT" << " ";
         myfile << i << " ";
         myfile << poses[i].x() << " " << poses[i].y() << " " << poses[i].z() << " ";
-        Vector quaternion = Rot3(R).quaternion();
-        myfile << quaternion(3) << " " << quaternion(2) << " " << quaternion(1)
-               << " " << quaternion(0);
+        Quaternion quaternion = Rot3(R).toQuaternion();
+        myfile << quaternion.x() << " " << quaternion.y() << " " << quaternion.z()
```

Comment:
Oh, wow, so this old code was incorrect? As I think this is not equivalent, right?

Reply:
Yes... according to the [g2o documentation](https://github.com/RainerKuemmerle/g2o/wiki/File-format-slam-3d#3d-robot-pose), the quaternion written to a `.g2o` file should be in the order of `qx,qy,qz,qw`,  while in the old code, it's actually written in the order of `qz,qy,qx,qw`(q3,q2,q1,q0). So I guess that's why the `saveG2oResult()` function in `timeShonanAveraging.cpp` was commented out because it didn't output a g2o file in the correct format? 

BTW, I've watched your lecture 'Pushing Factor Graphs beyond SLAM' before, which also mentioned the Shonan Averaging, very insightful talk!